service/src/main/kotlin/fi/espoo/evaka/incomestatement/IncomeStatementQueries.kt

```diff
@@ -0,0 +1,358 @@
+        .bindNullable("selfEmployedAttachments", null as Boolean?)
+        .bindNullable("selfEmployedEstimatedMonthlyIncome", null as Int?)
+        .bindNullable("selfEmployedIncomeStartDate", null as LocalDate?)
+        .bindNullable("selfEmployedIncomeEndDate", null as LocalDate?)
+        .bindNullable("limitedCompanyIncomeSource", null as IncomeSource?)
+        .bindNullable("partnership", null as Boolean?)
+        .bindNullable("lightEntrepreneur", null as Boolean?)
+        .bind("accountantName", "")
+        .bind("accountantAddress", "")
+        .bind("accountantPhone", "")
+        .bind("accountantEmail", "")
+        .bindNullable("student", null as Boolean?)
+        .bindNullable("alimonyPayer", null as Boolean?)
+        .bindNullable("otherInfo", null as String?)
+        .run {
+            when (body) {
+                is IncomeStatementBody.HighestFee ->
+                    bind("type", IncomeStatementType.HIGHEST_FEE)
+                is IncomeStatementBody.Income -> {
+                    bind("type", IncomeStatementType.INCOME)
+                        .run { if (body.gross != null) bindGross(body.gross) else this }
+                        .run { if (body.entrepreneur != null) bindEntrepreneur(body.entrepreneur) else this }
+                        .bind("student", body.student)
+                        .bind("alimonyPayer", body.alimonyPayer)
+                        .bind("otherInfo", body.otherInfo)
+                }
+            }
+        }
+
+private fun <This : SqlStatement<This>> SqlStatement<This>.bindGross(gross: Gross): This =
+    bind("grossIncomeSource", gross.incomeSource)
+        .run { if (gross.estimatedIncome != null) bindGrossEstimation(gross.estimatedIncome) else this }
+        .bind("grossOtherIncome", gross.otherIncome.toTypedArray())
+
+private fun <This : SqlStatement<This>> SqlStatement<This>.bindGrossEstimation(estimation: EstimatedIncome): This =
+    bind("grossEstimatedMonthlyIncome", estimation.estimatedMonthlyIncome)
+        .bind("grossIncomeStartDate", estimation.incomeStartDate)
+        .bindNullable("grossIncomeEndDate", estimation.incomeEndDate)
+
+private fun <This : SqlStatement<This>> SqlStatement<This>.bindEntrepreneur(entrepreneur: Entrepreneur): This =
+    run { if (entrepreneur.selfEmployed != null) bindSelfEmployed(entrepreneur.selfEmployed) else this }
+        .run { if (entrepreneur.limitedCompany != null) bindLimitedCompany(entrepreneur.limitedCompany) else this }
+        .bind("fullTime", entrepreneur.fullTime)
+        .bind("startOfEntrepreneurship", entrepreneur.startOfEntrepreneurship)
+        .bind("spouseWorksInCompany", entrepreneur.spouseWorksInCompany)
+        .bind("startupGrant", entrepreneur.startupGrant)
+        .bind("partnership", entrepreneur.partnership)
+        .bind("lightEntrepreneur", entrepreneur.lightEntrepreneur)
+        .run { if (entrepreneur.accountant != null) bindAccountant(entrepreneur.accountant) else this }
```

Comment:
I don't know if you looked into it, but `bindKotlin` could reduce a lot of this code, especially the places where an identically named field is bound.

Reply:
Didn't get `bindKotlin` to work 🤷